theano/tensor/var.py

```diff
@@ -470,19 +470,17 @@ def __getitem__(self, args):
 
         def check_bool(args_el):
             try:
-                if isinstance(args_el, (numpy.bool_, bool)) or \
-                   args_el.dtype == 'int8' or args_el.dtype == 'uint8':
-                    raise TypeError(('TensorType does not support boolean '
-                                     'mask for indexing such as tensor[x==0]. '
-                                     'Instead you can use non_zeros() such as '
-                                     'tensor[(x == 0).nonzeros()]. '
-                                     'If you are indexing on purpose with an '
-                                     'int8, please cast it to int16.'))
+                if (isinstance(args_el, (numpy.bool_, bool)) or
+                        args_el.dtype == 'bool'):
+                    raise TypeError('TensorType does not support boolean '
+                                    'mask for indexing such as tensor[x==0]. '
+                                    'Instead you can use non_zeros() such as '
+                                    'tensor[(x == 0).nonzeros()]. ')
```

Comment:
We could automatically index with `as_tensor_variable(args_el).nonzeros()` instead of raising an Exception, but it would be for another ticket.